test/check-machines-storage-pools

```diff
@@ -695,6 +695,59 @@ class TestMachinesStoragePools(VirtualMachinesCase):
             remove=True,
         ).execute()
 
+    def testGeneralCheckForPoolsDeletion(self):
+        b = self.browser
+        m = self.machine
+
+        args = self.createVm("subVmTest1")
+        m.execute("virsh vol-create-as images a 1M")
+        m.execute("virsh vol-create-as images z 1M")
+
+        self.login_and_go("/machines")
+        b.wait_in_text("body", "Virtual machines")
+
+        b.wait_in_text("#card-pf-storage-pools button .card-pf-title-link", "Storage pool")
+        b.click("#card-pf-storage-pools .pf-c-card__header button")
+        b.wait_visible("#storage-pools-listing")
+
+        # Check for deleting a pool whose disk is used by a VM
+        b.click("#pool-images-system-action-kebab button")
+        b.click("#pool-images-system-action-kebab a:contains(Delete)")
+        b.set_checked("#storage-pool-delete-volumes", True)
+        b.wait_in_text(".pf-c-backdrop .pf-c-helper-text__item-text", "Pool's volumes are used by")
+        b.wait_visible(".pf-c-backdrop footer button[aria-disabled=true]:contains(Delete)")
+
+        # Check "Cancel" button
+        b.click(".pf-c-backdrop footer button:contains(Cancel)")
+        b.wait_not_present(".pf-c-backdrop")
+
+        # Check "X" button of the dialog
+        b.click("#pool-images-system-action-kebab button")
+        b.click("#pool-images-system-action-kebab a:contains(Delete)")
+        b.click(".pf-c-backdrop button[aria-label=Close]")
+        b.wait_not_present(".pf-c-backdrop")
+
+        # Check that the error shown after the title, not the footer
+        # Delete the VM, or the deletion button will be disabled
+        m.execute("virsh destroy subVmTest1; virsh undefine subVmTest1")
+        # Check both command and UI that the VM is not shown
+        wait(lambda: "subVmTest1" not in m.execute("virsh list --all"))
+        b.click(".machines-listing-breadcrumb li a:contains(Virtual machines)")
+        b.wait_not_present("#vm-subVmTest1-system-name")
+        # Go back to the storage pool list
+        b.wait_in_text("#card-pf-storage-pools button .card-pf-title-link", "Storage pool")
+        b.click("#card-pf-storage-pools .pf-c-card__header button")
+        b.wait_visible("#storage-pools-listing")
+
+        m.execute(f'chattr +a {args["image"]}')
+        self.addCleanup(m.execute, f'chattr -a {args["image"]}')
+        b.click("#pool-images-system-action-kebab button")
+        b.click("#pool-images-system-action-kebab a:contains(Delete)")
+        b.set_checked("#storage-pool-delete-volumes", True)
+        b.click(".pf-c-backdrop footer button[aria-disabled=false]:contains(Delete)")
+        b.wait_in_text(".pf-c-backdrop header + div .pf-c-alert__description", "cannot unlink file")
```

Comment:
CHecking that the error appears in the dialog is already done in https://github.com/cockpit-project/cockpit-machines/blob/main/test/check-machines-storage-pools#L842 and in specific in this test https://github.com/cockpit-project/cockpit-machines/blob/main/test/check-machines-storage-pools#L954

If there is some extra check missing please extend that one instead.

Reply:
I think there are some different between them, as this check will check the other files which is not changed by chattr  will be deleted, although there is an error shown on the dialog and the dialog is not closed. Or do you think this behavior is not our expected?

Reply:
But this is not a discussion for the test now. :)

Reply:
Maybe we need a issue to track this behavior